src/main/java/com/mongodb/hibernate/jdbc/MongoResultSet.java

```diff
@@ -0,0 +1,295 @@
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+    }
+
+    @Override
+    public @Nullable Timestamp getTimestamp(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+    }
+
+    @Override
+    public @Nullable Timestamp getTimestamp(int columnIndex, Calendar cal) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+    }
+
+    @Override
+    public @Nullable BigDecimal getBigDecimal(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        BsonValue bsonValue = getBsonValue(columnIndex);
+        return bsonValue.isNull()
+                ? null
+                : bsonValue.asDecimal128().decimal128Value().bigDecimalValue();
+    }
+
+    @Override
+    public @Nullable Array getArray(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException();
+    }
+
+    @Override
+    public <T> @Nullable T getObject(int columnIndex, Class<T> type) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("In scope of Array / Struct implementation");
+    }
+
+    @Override
+    public ResultSetMetaData getMetaData() throws SQLException {
+        checkClosed();
+        return new MongoResultSetMetadata();
+    }
+
+    @Override
+    public int findColumn(String columnLabel) throws SQLException {
+        checkClosed();
+        var index = fieldNames.indexOf(columnLabel);
+        if (index < 0) {
+            throw new SQLException("Unknown column label: " + columnLabel);
+        }
+        return index + 1;
+    }
+
+    @Override
+    public boolean isWrapperFor(Class<?> iface) throws SQLException {
+        checkClosed();
+        return false;
+    }
+
+    private String getKey(int columnIndex) {
+        return fieldNames.get(columnIndex - 1);
+    }
+
+    private void checkClosed() throws SQLException {
+        if (closed) {
+            throw new SQLException("MongoResultSet has been closed");
+        }
+    }
+
+    private BsonValue getBsonValue(int columnIndex) throws SQLException {
+        lastReadColumnValue = assertNotNull(currentDocument).get(getKey(columnIndex), BsonNull.VALUE);
+        return lastReadColumnValue;
+    }
+
+    private void checkColumnIndex(int columnIndex) throws SQLException {
+        if (fieldNames.isEmpty()) {
+            throw new SQLException("No field exists");
+        }
```

Comment:
It seems ok not to have this if-check, since the ensuing will catch the issue?

Reply:
If the field names is empty, the following exception message would be confusing:

```
Invalid column index [5]; should be within [1, 0]
```
"should be within [1, 0]" would be really weird. The above check is simply to avoid that.

Reply:
Could we not change the wording? Something like:
`format("Invalid column index %d; cannot be under 1 or over the current number of fields, %d", columnIndex, fieldNames.size()));`

This will also log the invalid index in cases where fieldNames is empty.

Reply:
Good point! I changed as suggested. Will sync up similar changes in other JDBC classes in other PR.